packages/cysync-core/src/dialogs/DeployAccount/context/index.tsx

```diff
@@ -0,0 +1,436 @@
+import logger from '~/utils/logger';
+
+import { SummaryDialog, FinalMessage, DeviceAction } from '../Dialogs';
+
+export interface DeployAccountDialogContextInterface {
+  tabs: ITabs;
+  onNext: (tab?: number, dialog?: number) => void;
+  goTo: (tab: number, dialog?: number) => void;
+  onPrevious: () => void;
+  onSummaryDialogPrevious: () => void;
+  onClose: () => void;
+  onFinishDeployAccount: () => void;
+  onRetry: () => void;
+  error: any | undefined;
+  currentTab: number;
+  currentDialog: number;
+  selectedWallet: IWallet | undefined;
+  selectedAccount: IAccount | undefined;
+  selectedAccountParent: IAccount | undefined;
+  transaction: IPreparedTransaction | undefined;
+  setTransaction: (txn: IPreparedTransaction) => void;
+  prepare: () => Promise<void>;
+  isDeviceRequired: boolean;
+  deviceEvents: Record<number, boolean | undefined>;
+  startFlow: () => Promise<void>;
+  storedTransaction: ITransaction | undefined;
+  transactionLink: string | undefined;
+  getComputedFee: (
+    coinFamily: CoinFamily,
+    txn?: IPreparedTransaction,
+  ) => string;
+  defaultWalletId?: string;
+  defaultAccountId?: string;
+}
+
+export const DeployAccountDialogContext: Context<DeployAccountDialogContextInterface> =
+  createContext<DeployAccountDialogContextInterface>(
+    {} as DeployAccountDialogContextInterface,
+  );
+
+export interface DeployAccountDialogProps {
+  account: IAccount;
+  parentAccount?: IAccount;
+  wallet?: IWallet;
+  sendTxnData?: Record<string, string>;
+  isWalletConnectRequest?: boolean;
+}
+
+export interface DeployAccountDialogContextProviderProps
+  extends DeployAccountDialogProps {
+  children: ReactNode;
+}
+
+export const DeployAccountDialogProvider: FC<
+  DeployAccountDialogContextProviderProps
+> = ({
+  children,
+  wallet: selectedWallet,
+  account: selectedAccount,
+  parentAccount: selectedAccountParent,
+  sendTxnData,
+  isWalletConnectRequest,
+}) => {
+  const lang = useAppSelector(selectLanguage);
+  const dispatch = useAppDispatch();
+  const deviceRequiredDialogsMap: Record<number, number[] | undefined> =
+    useMemo(
+      () => ({
+        1: [0],
+        2: [0],
+      }),
+      [],
+    );
+
+  const [error, setError] = useState<any | undefined>();
+  const [signedTransaction, setSignedTransaction] = useState<
+    string | undefined
+  >();
+  const [storedTransaction, setStoredTransaction] = useState<
+    ITransaction | undefined
+  >();
+  const [transactionLink, setTransactionLink] = useState<string | undefined>();
+  const [transaction, setTransaction, transactionRef] = useStateWithRef<
+    IPreparedTransaction | undefined
+  >(undefined);
+
+  const coinSupport = useRef<CoinSupport | undefined>();
+  const [deviceEvents, setDeviceEvents] = useState<
+    Record<number, boolean | undefined>
+  >({});
+  const { connection } = useDevice();
+  const flowSubscription = useRef<Subscription | undefined>();
+  const { rejectCallRequest, callRequestData } = useWalletConnect();
```

Comment:
I don't think we need wallet connect features here

Reply:
we need to reject the request if the user closes the dialog